idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -533,127 +423,84 @@ def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
-            # 	blk.outlet.flow_mol[t]*blk.outlet.enth_mol[t])
-
-        # Deactivate special constraints
-        blk.eff_isen_eqn.deactivate()
-        blk.Pratio_con.deactivate()
-        blk.kappaT_con.deactivate()
-        blk.c0_con.deactivate()
-        blk.Ma_con.deactivate()
-        blk.rspeed_con.deactivate()
-        blk.eff_p_v_cons.deactivate()
-        blk.polytropic_correlation.deactivate()
-        blk.delta_enth_polytropic_con.deactivate()
-        blk.mass_flow_coeff_eqn.deactivate()
-        blk.psi_3_eqn.deactivate()
-        blk.psi_s_eqn.deactivate()
-        blk.psi_s_stage_eqn.deactivate()
-
-        # Fix variables
-        blk.kappaT.fix()
-        blk.c0.fix()
-        blk.Ma.fix()
-        blk.rspeed.fix()
-        blk.impeller_work_coeff.fix()
-        blk.delta_enth_polytropic.fix()
-        blk.mass_flow_coeff.fix()
-        blk.psi_3.fix()
-        blk.psi_s.fix()
-
-        opt = SolverFactory(solver)
-        opt.options = optarg
-
-        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-            res = opt.solve(blk, tee=slc.tee)
-
-        init_log.info_high(
-                "Initialization Step 1 {}.".format(idaeslog.condition(res))
-            )
-
+            if self.outlet.pressure[t].fixed:
+                self.ratioP[t] = pyo.value(
+                    self.outlet.pressure[t] / self.inlet.pressure[t])
+            elif self.control_volume.deltaP[t].fixed:
+                self.ratioP[t] = pyo.value(
+                    (self.control_volume.deltaP[t] + self.inlet.pressure[t]
+                     ) / self.inlet.pressure[t])
+            elif self.ratioP[t].fixed:
+                self.outlet.pressure[t] = pyo.value(
+                    self.ratioP[t] * self.inlet.pressure[t])
+            else:
+                if value(self.ratioP[t]) is None or value(
+                        self.ratioP[t]) < 1.01:
+                    self.ratioP[t].fix(1.5)
+                else:
+                    self.ratioP[t].fix()
+                unfix_ratioP[t] = True
+
+        # set list of constraints
+        constraint_list = [
+            "impeller_work_coeff_eqn",
+            "polytropic_head_coeff_vaned_diffuser_eqn",
+            "polytropic_head_coeff_eqn",
+            "polytropic_efficiency_eqn",
+            "Ma_con",
+            "rspeed_con",
+            "eff_p_v_cons",
+            "polytropic_correlation",
+            "delta_enth_polytropic_con",
+            "mass_flow_coeff_eqn",
+            "psi_3_eqn",
+            "psi_s_eqn"]
+
+        # deactivate unit model level constraints
+        for c in constraint_list:
+            self.component(c).deactivate()
+
+        state_args = {
+            'flow_mol': self.vapor_inlet.flow_mol[0].value,
+            'pressure': self.vapor_inlet.pressure[0].value,
+            'enth_mol': self.vapor_inlet.enth_mol[0].value
+        }
+
+        flags = self.control_volume.initialize(
```

Comment:
This step seems redundant - the compressor initialization (next step) should do this for you anyway, Also, passing the `state_args` also appears redundant, as you are just using the inlet conditions, which is the default for the compressor initialization anyway.